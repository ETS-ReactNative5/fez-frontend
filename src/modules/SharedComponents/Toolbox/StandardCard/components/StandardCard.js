import React, {Component} from 'react';
import PropTypes from 'prop-types';

import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import {HelpIcon} from '../../HelpDrawer';
import {withStyles} from '@material-ui/core/styles';

const styles = theme => ({
    card: {
        // marginBottom: 24
    },
    cardHeaderPurple: {
        margin: '12px 12px 0 12px',
        color: theme.palette.white.main,
        backgroundColor: theme.palette.primary.main,
        '& h2': {
            paddingLeft: 12
        },
    },
    cardHeader: {
        margin: '24px 12px 0 12px',
        '& h2': {
            paddingLeft: 12
        },
    },
    cardContent: {
        fontWeight: theme.typography.fontWeightRegular
    },
    noPadding: {
        padding: 0,
    },
    fullHeight: {
        border: '10px solid red',
        height: '100%'
    }
});

class Cards extends Component {
    static propTypes = {
        title: PropTypes.any,
        darkHeader: PropTypes.bool,
        fullHeight: PropTypes.bool,
        noPadding: PropTypes.bool,
        noHeader: PropTypes.bool,
        children: PropTypes.any,
        classes: PropTypes.object,
        help: PropTypes.object,
        customBackgroundColor: PropTypes.any,
        customTitleColor: PropTypes.any
    };
    render() {
        const {classes, title, help, children, darkHeader} = this.props;
        const customBG = !!this.props.customBackgroundColor ? {backgroundColor: this.props.customBackgroundColor} : null;
        const customTitle = !!this.props.customTitleColor ? {color: this.props.customTitleColor} : null;
        const fullHeight = !!this.props.fullHeight ? {height: '100%'} : null;
        return (
            <Card className={classes.card} style={{...customBG, ...customTitle, ...fullHeight}}>
                {
                    !this.props.noHeader &&
                    <Grid container spacing={24}>
                        {
                            title ?
                                <Grid item xs className={darkHeader && classes.cardHeaderPurple || classes.cardHeader}>
                                    <Typography variant={'title'} color={'inherit'}>{title}</Typography>
                                </Grid>
                                :
                                <Grid item xs/>
                        }
                        {
                            help && help.text &&
                            <Grid item>
                                <HelpIcon {...help}/>
                            </Grid>
                        }
                    </Grid>
                }
                <CardContent classes={this.props.noPadding ? {root: classes.noPadding} : {}}>
                    <Grid container spacing={24}>
                        <Grid item xs={12} className={classes.cardContent}>
                            {children}
                        </Grid>
                    </Grid>
                </CardContent>
            </Card>
        );
    }
}

const StyledCard = withStyles(styles, {withTheme: true})(Cards);
const StandardCard = (props) => <StyledCard {...props}/>;
export default StandardCard;
