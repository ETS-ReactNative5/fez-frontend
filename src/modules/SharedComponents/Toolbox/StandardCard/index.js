export {default as StandardCard} from './components/StandardCard';
