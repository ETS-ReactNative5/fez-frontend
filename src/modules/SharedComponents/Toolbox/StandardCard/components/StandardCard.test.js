import {Cards} from './StandardCard';

function setup(testProps, isShallow = true) {
    const props = {
        title: 'card title',
        classes: {
            card: 'testClass'
        },
        ...testProps
    };
    return getElement(Cards, props, isShallow);
}

describe('StandardCard component', () => {

    it('renders with title and no help icon', () => {
        const wrapper = setup({});
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with title and help button', () => {
        const wrapper = setup({
            help: {
                title: 'help',
                text: 'help text',
                buttonLabel: 'OK'
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('renders with custom colours and full height', () => {
        const wrapper = setup({
            customBackgroundColor: '#fcc',
            customTitleColor: '#111',
            fullHeight: true,
            noPadding: true,
            darkHeader: true,
            classes: {
                cardHeaderPurple: 'purple',
                cardContentNoPadding: 'no-padding'
            }
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
