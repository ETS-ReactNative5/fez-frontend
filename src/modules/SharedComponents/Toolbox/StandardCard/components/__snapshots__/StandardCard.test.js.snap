// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component renders with custom colours and full height 1`] = `
<WithStyles(Card)
  className="undefined StandardCard"
  style={
    Object {
      "backgroundColor": "#fcc",
      "color": "#111",
      "height": "100%",
    }
  }
>
  <WithStyles(CardHeader)
    action={false}
    classes={
      Object {
        "root": "#555",
      }
    }
    style={
      Object {
        "backgroundColor": "#ccc",
      }
    }
    title="card title"
    titleTypographyProps={
      Object {
        "color": "inherit",
        "component": "h3",
        "variant": "h5",
      }
    }
  />
  <WithStyles(CardContent)
    className="no-padding"
  />
</WithStyles(Card)>
`;

exports[`Cards component renders with square top and accent header 1`] = `
<WithStyles(Card)
  className="undefined StandardCard"
  style={Object {}}
>
  <WithStyles(CardHeader)
    action={false}
    classes={
      Object {
        "root": "#333",
      }
    }
    style={
      Object {
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
      }
    }
    title="card title"
    titleTypographyProps={
      Object {
        "color": "inherit",
        "component": "h3",
        "variant": "h5",
      }
    }
  />
  <WithStyles(CardContent)
    className=""
  />
</WithStyles(Card)>
`;

exports[`Cards component renders with title and help button 1`] = `
<WithStyles(Card)
  className="testClass StandardCard"
  style={Object {}}
>
  <WithStyles(CardHeader)
    action={
      <Memo(Connect(WithStyles(HelpIcon)))
        buttonLabel="OK"
        text="help text"
        title="help"
      />
    }
    classes={
      Object {
        "root": undefined,
      }
    }
    style={Object {}}
    title="card title"
    titleTypographyProps={
      Object {
        "color": "inherit",
        "component": "h3",
        "variant": "h5",
      }
    }
  />
  <WithStyles(CardContent)
    className=""
  />
</WithStyles(Card)>
`;

exports[`Cards component renders with title and no help icon 1`] = `
<WithStyles(Card)
  className="testClass StandardCard"
  style={Object {}}
>
  <WithStyles(CardHeader)
    action={false}
    classes={
      Object {
        "root": undefined,
      }
    }
    style={Object {}}
    title="card title"
    titleTypographyProps={
      Object {
        "color": "inherit",
        "component": "h3",
        "variant": "h5",
      }
    }
  />
  <WithStyles(CardContent)
    className=""
  />
</WithStyles(Card)>
`;

exports[`StandardCard component should render StyledCard with same props 1`] = `
<WithStyles(Cards)
  test1="test1value"
  test2="test2value"
/>
`;
